src/Core/ObjectRegistry.php

```diff
@@ -180,10 +181,10 @@ protected function _checkDuplicate(string $name, array $config): void
      * This method should construct and do any other initialization logic
      * required.
      *
-     * @param string|object $class The class to build.
+     * @param string|TObject $class The class to build.
      * @param string $alias The alias of the object.
      * @param array $config The Configuration settings for construction
-     * @return object
+     * @return TObject
```

Comment:
Do IDEs also understand this?

Reply:
Dunno. People can submit issue for their IDE of choice if it doesn't.

Reply:
Dunno. People can submit issue for their IDE of choice if it doesn't.